client/src/components/forms/MailingList/index.tsx

```diff
@@ -1,11 +1,141 @@
+    height: LOGO_SIZE,
+    marginBottom: theme.spacing(1)
+  },
+  logo: {
+    maxWidth: '100%',
+    maxHeight: '100%',
+    borderRadius: LOGO_SIZE / 2
+  },
+  slider: {
+    display: 'flex',
+    width: 400,
+    marginBottom: theme.spacing(2),
+    boxSizing: 'border-box',
+    position: 'relative',
+    overflow: 'hidden',
+    transition: TRANSITION_TIME
+  },
+  slide: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    minWidth: '100%',
+    minHeight: 100,
+    transition: TRANSITION_TIME
+  },
+  buttonContainer: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  button: {
+    transition: TRANSITION_TIME
+  }
+});
+
+const isFirstSlide = (translateValue: number) => translateValue === 0;
+
+const isLastSlide = (
+  translateValue: number
+) => translateValue === (-100 * (NUMBER_OF_SLIDES - 1));
+
+function MailingListForm({ classes }: WithStyles<typeof styles>) {
+  const [email, setEmail] = useState('');
+  const [events, setEvents] = useState<EVENT[]>([]);
+  const [feedback, setFeedback] = useState('');
+  const [translateValue, setTranslateValue] = useState(0);
+
+  const goLeft = () => {
+    setTranslateValue(translateValue + 100);
+  }
+
+  const goRight = () => {
+    setTranslateValue(translateValue - 100);
+  }
 
-function MailingListForm () {
   return (
-    <div>
-        Hello
+    <div className={classes.container}>
+      <div className={classes.logoContainer}>
+        <img
+          className={classes.logo}
+          src='/assets/logo.jpg'
+          alt='ACS Logo'
+        />
+      </div>
+
+      <div className={classes.slider}>
+        <div
+          className={classes.slide}
+          style={{ transform: `translateX(${translateValue}%)`}}
+        >
+          <EmailSlide
+            email={email}
+            setEmail={setEmail}
+          />
+        </div>
+        <div
+          className={classes.slide}
+          style={{ transform: `translateX(${translateValue}%)`}}
+        >
+          <FeedbackSlide
+            events={events}
+            setEvents={setEvents}
+            feedback={feedback}
+            setFeedback={setFeedback}
+          />
+        </div>
+        <div
+          className={classes.slide}
+          style={{ transform: `translateX(${translateValue}%)`}}
+        >
+          <SocialSlide />
+        </div>
+      </div>
+
+      <div
+        className={classes.buttonContainer}
+        style={!isFirstSlide(translateValue) ? {width: '100%'} : {}}
+      >
+        {!isFirstSlide(translateValue) && !isLastSlide(translateValue) && (
+          <Button onClick={goLeft} className={classes.button}>
+            <NavigateBeforeIcon />
+            Back
+          </Button>
+        )}
+        {!isLastSlide(translateValue) && (
+          <Button onClick={goRight} className={classes.button}>
+            {translateValue === (-100 * (NUMBER_OF_SLIDES - 2)) ? 'Submit' : 'Next'}
```

Comment:
I would've preferred keeping `currentSlide` in the state since it's more meaningful and then calculating the `translateValue` but this works

Reply:
Makes sense. I'll probably update

Reply:
Actually I can't because the slides go like

`0` `-100` `-200`

If the values were positive, I'd be more inclined to use `currentSlide`